resources/assets/lib/chat/conversation-list.tsx

```diff
@@ -1,33 +1,47 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
-import { inject, observer } from 'mobx-react';
-import Channel from 'models/chat/channel';
+import { computed } from 'mobx';
+import { observer } from 'mobx-react';
+import core from 'osu-core-singleton';
 import * as React from 'react';
 import ConversationListItem from './conversation-list-item';
 
-@inject('dataStore')
 @observer
-export default class ConversationList extends React.Component<any> {
+export default class ConversationList extends React.Component {
+  @computed
+  private get nonPmChannels() {
+    return core.dataStore.channelStore.nonPmChannels;
+  }
+
+  @computed
+  private get pmChannels() {
+    return core.dataStore.channelStore.pmChannels;
+  }
+
   render(): React.ReactNode {
-    const nonPmChannels: Channel[] = this.props.dataStore.channelStore.nonPmChannels;
-    const pmChannels: Channel[] = this.props.dataStore.channelStore.pmChannels;
-    const conversationList: React.ReactNode[] = [];
-
-    nonPmChannels.forEach((conversation) => {
-      conversationList.push(<ConversationListItem key={conversation.channelId} channelId={conversation.channelId} />);
-    });
-
-    if (nonPmChannels.length > 0 && pmChannels.length > 0) {
-      conversationList.push(
-        <div key='separator' className='chat-conversation-list-separator' />,
-      );
-    }
+    return (
+      <div className='chat-conversation-list'>
+        {this.renderNonPmChannels()}
+        {this.renderSeparator()}
+        {this.renderPmChannels()}
+      </div>
+    );
+  }
 
-    pmChannels.forEach((conversation) => {
-      conversationList.push(<ConversationListItem key={conversation.channelId} channelId={conversation.channelId} />);
-    });
+  private renderNonPmChannels() {
+    return this.nonPmChannels.map((conversation) => <ConversationListItem key={conversation.channelId} channelId={conversation.channelId} />);
+  }
+
+  private renderPmChannels() {
+    return this.pmChannels.map((conversation) => <ConversationListItem key={conversation.channelId} channelId={conversation.channelId} />);
```

Comment:
the actual functions seem rather same so `renderChannel` instead and map from main `render` function?